### PR TITLE
Define additional filled colors in interfaces.txt

### PR DESCRIPTION
**Pull request by tehhowch**
- Further support for modified interfaces / reskinning, such as @comnom 's "Pug HUD" ( https://github.com/comnom/es-huds )

### CODE REVIEW

File: data/interfaces.txt [hunk 1]
```diff
@@ -17,6 +17,17 @@ color "medium" .5 .5 .5 0.
 color "dim" .25 .25 .25 0.
 color "faint" .1 .1 .1 0.
 
+color "panel background" .1 .1 .1 1.
+color "dialog backdrop" 0. 0. 0. .7
+color "conversation background" .125 .125 .125 1.
+color "map sidepanel background" .125 .125 .125 1.
+color "shop sidepanel background" .2 .2 .2 1.
+color "shop sidepanel footer" .3 .3 .3 1.
```
Comment (endless-sky) [inline data/interfaces.txt]: I think "side panel" should be two words. "Sidebar" is a well-established word, but "sidepanel" is not.

Comment (tehhowch) [reply data/interfaces.txt]: Perhaps these should be sidebar then instead of sidepanel? Name suggestions are always welcomed

**Comment by tehhowch** [comment]
I've added some comments to the colors definition section to indicate what the colors are used for (so that the name alone doesn't need to convey what the color is for).

File: source/MissionPanel.cpp [hunk 2]
```diff
@@ -506,9 +506,9 @@ void MissionPanel::DrawMissionSystem(const Mission &mission, const Color &color)
 Point MissionPanel::DrawPanel(Point pos, const string &label, int entries) const
 {
 	const Font &font = FontSet::Get(14);
-	Color back(.125, 1.);
-	Color unselected = *GameData::Colors().Get("medium");
-	Color selected = *GameData::Colors().Get("bright");
+	const Color &back = *GameData::Colors().Get("map sidepanel background");
```
Comment (endless-sky) [inline source/MissionPanel.cpp]: This name needs to be updated ("side panel")

File: source/ShopPanel.cpp [hunk 3]
```diff
@@ -148,19 +148,19 @@ void ShopPanel::Draw()
...
 	FillShader::Fill(
 		Point(Screen::Right() - SIDE_WIDTH / 2, 0.),
 		Point(SIDE_WIDTH, Screen::Height()),
-		Color(.1, 1.));
+		*GameData::Colors().Get("panel background"));
 	FillShader::Fill(
 		Point(Screen::Right() - SIDE_WIDTH, 0.),
 		Point(1, Screen::Height()),
-		Color(.2, 1.));
+		*GameData::Colors().Get("shop sidepanel background"));
```
Comment (endless-sky) [inline source/ShopPanel.cpp]: Same here.

Comment (tehhowch) [reply source/ShopPanel.cpp]: that's an oops. :laughing:

File: source/ShopPanel.cpp [hunk 4]
```diff
@@ -262,14 +262,15 @@ void ShopPanel::DrawButtons()
 {
 	// The last 70 pixels on the end of the side panel are for the buttons:
 	Point buttonSize(SIDE_WIDTH, BUTTON_HEIGHT);
-	FillShader::Fill(Screen::BottomRight() - .5 * buttonSize, buttonSize, Color(.2, 1.));
+	FillShader::Fill(Screen::BottomRight() - .5 * buttonSize, buttonSize, *GameData::Colors().Get("shop sidepanel background"));
```
Comment (endless-sky) [inline source/ShopPanel.cpp]: And, here and below.

File: source/MapSalesPanel.cpp [hunk 5]
```diff
@@ -269,7 +269,7 @@ void MapSalesPanel::DrawInfo() const
 			width += box->Width() + compareInfo.PanelWidth();
 		}
 		
-		Color back(.125, 1.);
+		const Color &back = *GameData::Colors().Get("map sidepanel background");
```
Comment (endless-sky) [inline source/MapSalesPanel.cpp]: And these two.

**Comment by tehhowch** [comment]
@endless-sky got 'em all

**Pull request merged**